app/views/admin/accounting/transactions/_modal_content.slim

```diff
@@ -8,7 +8,7 @@ div.modal-header
       = @transaction.description
 
 div.modal-body
-  - unless @transaction.new_record? || @transaction.type?("interest")
+  - if !@transaction.new_record? && policy(@transaction).update?
```

Comment:
Nice improvement! You can also do `persisted?` to avoid the negation, but this is fine.